dbprocessing/Diskfile.py

```diff
@@ -144,13 +144,17 @@ def calcDigest(infile):
     :rtype: str
 
     """
-    m = hashlib.sha1()
     try:
-        with open(infile, 'rb') as f:
-            m.update(f.read())
+        if not os.path.isfile(infile):
+            raise IOError
+        cmd = "sha1sum " + infile
+        p = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, err = p.communicate()
+        p.wait()
     except IOError:
         raise DigestError("File not found: {0}".format(infile))
         
-    DBlogging.dblogger.debug("digest calculated: {0}, file: {1} ".format(m.hexdigest(), infile))
+    result = out.split()[0].decode('ascii')
+    DBlogging.dblogger.debug("digest calculated: {0}, file: {1} ".format(result, infile))
 
-    return m.hexdigest()
+    return result
```

Comment:
It's very slow for me with python on my machine here.  You might have more memory.  Our files are easily 200Gb.    You said the first time it took 15.5s, we only call this once.   

If you want to put back "hashlib" I will run a patch on this code after cloning, or use a separated branch where I merge your master using "theirs" and "ours".

I will try to do more tests, but try with a bigger file than 4.5Gb.  I might be a memory issues.  I think that 'sha1sum' should be faster since it's compiled.

Reply:
The big file was what I needed. Let me know when you've checked the samples I provided you and we can get that integrated.

Reply:
Okay, based on our conversation new version uses the hashlib calculation in chunks, will be pushed later (hopefully later today.)